File: js/18110.js
const fs = require('fs');

const inputs = fs.readFileSync('example.txt').toString().trim().split('\n');

const n = +inputs[0];
const arr = [];

if (n == 0) {
  console.log(0);
} else {
  for (let i = 1; i <= n; i++) {
    arr.push(+inputs[i]);
  }
  arr.sort((a, b) => a - b);

  const excludedNum = Math.round(n * 0.15);

  const answer = Math.round(
    arr.slice(excludedNum, n - excludedNum).reduce((a, b) => a + b, 0) /
      (n - 2 * excludedNum)
  );

  console.log(answer);
}

File: c++/18110.cpp
#include <algorithm>
#include <cmath>
#include <iostream>
#include <vector>

using namespace std;

vector<int> v;

int main() {
  freopen("example.txt", "r", stdin);
  int n;
  cin >> n;

  for (int i = 0; i < n; i++) {
    int a;
    cin >> a;
    v.push_back(a);
  }

  sort(v.begin(), v.end());

  if (n == 0) {
    cout << 0;
  } else {
    int excludedNum = round(n * 0.15);
    float sum = 0;
    for (int i = excludedNum; i < n - excludedNum; i++) {
      sum += v[i];
    }
    int answer = round(sum / (n - 2 * excludedNum));
    cout << answer;
  }

  return 0;
}